File: android/src/main/kotlin/io/openim/flutter_openim_sdk_ffi/listener/OnSignalingListener.kt
package io.openim.flutter_openim_sdk_ffi.listener

import io.openim.flutter_openim_sdk_ffi.models.CustomSignalingInfo
import io.openim.flutter_openim_sdk_ffi.models.MeetingStreamEvent
import io.openim.flutter_openim_sdk_ffi.models.RoomCallingInfo
import io.openim.flutter_openim_sdk_ffi.models.SignalingInfo

interface OnSignalingListener {
    fun onInvitationCancelled(s: SignalingInfo?)
    fun onInvitationTimeout(s: SignalingInfo?)
    fun onInviteeAccepted(s: SignalingInfo?)
    fun onInviteeAcceptedByOtherDevice(s: SignalingInfo?)
    fun onInviteeRejected(s: SignalingInfo?)
    fun onInviteeRejectedByOtherDevice(s: SignalingInfo?)
    fun onReceiveNewInvitation(s: SignalingInfo?)
    fun onHangup(s: SignalingInfo?)
    fun onRoomParticipantConnected(s: RoomCallingInfo?)
    fun onRoomParticipantDisconnected(s: RoomCallingInfo?)
    fun onMeetingStreamChanged(e: MeetingStreamEvent?)
    fun onReceiveCustomSignal(s: CustomSignalingInfo?)
}

File: ios/Classes/listener/OnSignalingListener.swift
protocol OnSignalingListener {
    func onInvitationCancelled(_ info: SignalingInfo?)
    func onInvitationTimeout(_ info: SignalingInfo?)
    func onInviteeAccepted(_ info: SignalingInfo?)
    func onInviteeAcceptedByOtherDevice(_ info: SignalingInfo?)
    func onInviteeRejected(_ info: SignalingInfo?)
    func onInviteeRejectedByOtherDevice(_ info: SignalingInfo?)
    func onReceiveNewInvitation(_ info: SignalingInfo?)
    func onHangup(_ info: SignalingInfo?)
    func onRoomParticipantConnected(_ info: RoomCallingInfo?)
    func onRoomParticipantDisconnected(_ info: RoomCallingInfo?)
    func onMeetingStreamChanged(_ event: MeetingStreamEvent?)
    func onReceiveCustomSignal(_ info: CustomSignalingInfo?)
}